solr/modules/s3-repository/src/test/org/apache/solr/s3/S3PathsTest.java

```diff
@@ -48,7 +48,7 @@ public void testFiles() throws S3Exception {
   public void testDirectory() throws S3Exception {
 
     client.createDirectory("/simple-directory");
-    assertTrue(client.pathExists("/simple-directory"));
+    assertFalse("Dir path needs a trailing slash", client.pathExists("/simple-directory"));
```

Comment:
Hmmm so it appears adobe mock no-longer supports this, but I'm pretty sure that S3 supports it... Kind of annoying.

Not sure what this does with regards to test coverage... As long as the integration test works, I imagine we would be fine here....

I'll try to run the nightlies on the module and see how it goes.

Reply:
so I haven't looked in detail, but the s3 client did upgrade so maybe this is a client side AWS change?

Reply:
So the integration tests work when I changed one line (which I have pushed), so this _should_ be good, though we have seen other strange errors that the integration tests didn't address. But overall I think we should be good here. directories are generally always treated as directories in the actual backup code...

Reply:
The nightlies pass across all implementations but hdfs. I don't think the hdfs failures were related to the change (the tests wouldn't even start)

Reply:
> I don't think the hdfs failures were related to the change (the tests wouldn't even start)

ut oh... I'll take a closer look at this since I just upgrade Hadoop to 3.3.5... I haven't seen any Jenkins failures related to it though.

Reply:
hmmm I don't see the same issue w/ hdfs tests running `./gradlew test -Ptests.filter="@Nightly and not(@awaitsfix)"`

Reply:
Also https://ci-builds.apache.org/job/Solr/job/Solr-NightlyTests-main/748/ passed w/o issue for HDFS so not sure what you see :/